packages/fdc3-agent-proxy/src/messaging/AbstractMessaging.ts

```diff
@@ -102,15 +112,18 @@ export abstract class AbstractMessaging implements Messaging {
         this.implementationMetadata = null;
     }
 
-    getImplementationMetadata(): Promise<ImplementationMetadata> {
-        return Promise.resolve(this.implementationMetadata!!)
+    async getImplementationMetadata(): Promise<ImplementationMetadata> {
+        if(this.implementationMetadata === null){
+            throw new Error("Unconnected. No metadata available.")
+        }
+        return this.implementationMetadata
     }
 
     private async exchangeValidationWithId<X>(message: any, connectionAttemptUuid: string): Promise<X> {
```

Comment:
I'm moving the identity validation into the Loaders and out of here in another branch. Its part of the connection flow for Web and should be separated from messaging (that could be used with a different connection).